backend/app/endpoints/squads.py

```diff
@@ -1,170 +1,235 @@
     squad_repository: SquadRepository = Depends(),
+    squad: Squad = Depends(get_owned_squad),
 ):
-    squad = squad_repository.get_squad(id)
-
-    if not squad:
-        raise HTTPException(status_code=404, detail="Squad not found")
-
-    if squad.owner_id != current_user.id:
-        raise HTTPException(status_code=403, detail="You are not the owner")
-
     return squad_repository.add_squad_avatar(squad, avatar)
 
 
-@router.post("/")
+@router.post("/", response_model=SquadResponse)
 def create_squad(
-    request: SquadUpsertRequest,
+    payload: SquadUpsertRequest,
     current_user: User = Depends(get_current_user),
     squad_repository: SquadRepository = Depends(),
 ):
-    squad = Squad(**request.model_dump(), owner_id=current_user.id)
-
-    if squad_repository.get_count_by_name(squad.name):
+    if squad_repository.get_count_by_name(payload.name):
         raise HTTPException(status_code=400, detail="Squad already exists")
 
     if (
-        squad_repository.get_count_by_owner(squad.owner_id)
+        squad_repository.get_count_by_owner(current_user.id)
         >= SquadPermissions.MAX_OWNED
     ):
         raise HTTPException(
             status_code=400,
             detail=f"You can't create more than {SquadPermissions.MAX_OWNED} squads",
         )
 
+    squad = Squad(
+        **payload.model_dump(), owner_id=current_user.id, members=[current_user]
+    )
+
     return squad_repository.upsert_squad(squad)
 
 
-@router.put("/{id}")
+@router.put("/{squad_id}")
 def update_squad(
-    id: int,
-    request: SquadUpsertRequest,
-    current_user: User = Depends(get_current_user),
+    payload: SquadUpdateRequest,
     squad_repository: SquadRepository = Depends(),
+    squad: Squad = Depends(get_owned_squad),
 ):
-    squad = Squad(**request.model_dump(), owner_id=current_user.id)
+    return squad_repository.update_squad(squad, payload)
 
-    if not squad_repository.get_squad(id):
-        raise HTTPException(status_code=404, detail="Squad not found")
 
-    if squad_repository.get_count_by_name(squad.name):
-        raise HTTPException(status_code=400, detail="Squad already exists")
+@router.post("/{squad_id}/invites")
+def invite_user(
+    payload: SquadInvitationRequest,
+    squad: Squad = Depends(get_owned_squad),
+    squad_repository: SquadRepository = Depends(),
+    user_repository: UserRepository = Depends(),
+):
+    user = user_repository.get_by_email(payload.email)
 
-    return squad_repository.upsert_squad(squad)
+    if not user:
+        raise HTTPException(status_code=404, detail="User not found")
 
+    if user in squad.members:
+        raise HTTPException(status_code=400, detail="User is already a member")
 
-@router.get("/{id}/invitations", response_model=List[UserResponse])
-def get_invited_users(
-    id: int,
-    _: User = Depends(get_current_user),
+    if user in squad.invitations:
+        raise HTTPException(status_code=400, detail="User already invited")
```

Comment:
No need to raise an error. Possibly, later here will be a notification service.
Update incitation `updated_at` and return `return {"message": "User invited"}`

```suggestion
        # touch `updated_at` field here
        return {"message": "User invited"}
```

Reply:
updated_at will be automatically updated. removed the if statement.